docs/source/design/ui_design.rst

```diff
@@ -0,0 +1,328 @@
+package. Note that, the input for the ``qcng.compute`` function is a Python
+dictionary with a schema defined by QCElemental.
+
+NWChemEx
+========
+
+In NWChemEx, running an SCF calculation for a hydrogen molecule is made easy
+through the use of a Python function with named arguments, as shown below.
+
+.. code-block:: python
+
+    import nwchemex as nwx 
+    energy = nwx.calculate_scf_energy(molecule = 'H 0. 0. 0. \n H 0. 0. 1.', basis = 'sto-3g')
+
+While the example shows only the two required arguments for the ``nwx.calculate_scf_energy()``
+function, the whole function signature is given below: 
+
+.. code-block:: python
+
+    def calculate_scf_energy(molecule: Union[str, chemist.Molecule], basis:
+    Union[str, simde.type.ao_space], spin: int = 1, max_iterations: int = 50,
+    scf_thresh: float = 1e-10, options: Union[Dict, dataclass] = None, **kwargs) -> float:
+
+In this function, the only required arguments are ``molecule``, which can either
+be a Python ``string`` or a ``chemist.Molecule`` object and ``basis``, which can
+either be a a Python ``string`` or a ``simde.type.ao_space``. Additional SCF
+related options are also explicitly defined in the function with reasonable
+default values. While the number of options are limited for SCF, for correlated
+methods more options may need to be defined on top of the SCF related options.
+One solution to avoid a long list of arguments is to provide additional options
+through ``**kwargs`` argument at the end. However, this would require users to
+know the correct keywords. To help our users to figure out the available
+keywords, we also provide the ``options`` argument, which is an object that
+holds all the possible options with reasonable default values. Our users can
+grab this object and figure out the available options and their default values
+and modify them as they wish. 
+
+Similarly, ``calculate_X_energy`` functions will also be provided for all other
+quantum chemistry methods supported by NWChemEx. Users may also request other
+properties such as the gradients and the wave function using any of the methods. 
+These can be obtained by ``calculate_X_gradients`` and
+``calculate_X_wavefunction``. It should be noted that, caching mechanism of
+NWChemEx avoids any recomputation if the requested quantity is already calculated.
+
+Our UI also provides a more general ``nwx.calculate`` function, where the method
+and the return types can be specified in the ``options`` explicitly or provided
+as key-value arguments in the function call.
+
+.. code-block:: python
+
+    def calculate(molecule: Union[str, chemist.Molecule], method: str, basis:
+    Union[str, simde.type.ao_space], return_energy: Bool = True, options:
+    Union[Dict, dataclass] = None, **kwargs):
+
+Using this function, a user can run any method and request different return
+types either using ``options`` or by setting these arguments explicitly.
+
```

Comment:
I really think you should lead with this and then discuss `calculate_scf_energy` as a simplification. That said, it's not even clear to me that having `calculate_scf_energy` etc. are worth it. They basically just hard-code two parameters: the method and the derivative order.

Reply:
I agree for promoting the `nwx.calculate` over the method specific ones. Making "energy" and "scf" inputs would be basically the same number of keystrokes. And I think that "method" and "return quantity" should be explicit inputs, not part of the `options`.

Reply:
A lot of the remaining conversations really are focused on the details of the method-specific `compute_xxx_yyy` functions. Honestly, I don't see these functions as being on the critical path as, IMHO, the `nwx.compute` function is pretty straightforward already.

So, in order to move forward, what I'd suggest is having the current design focus exclusively on the `nwx.compute` function. We can then add a section titled something like "future considerations" and move the discussion of method-specific UIs there with a disclaimer that the provided APIs are illustrative and subject to change.

Reply:
Basically it allows us to punt those discussions to a later time.